File: common/do_srgb_gam.m
function out = do_srgb_gam(in)
% in = 0-1.0, out=0-1.0
    srgb_lut = get_srgb_lut();
    out = interp1_clip(linspace(0, 1, 256), srgb_lut, in);
end

File: common/do_srgb_gam.py

import numpy as np
from .get_srgb_lut import get_srgb_lut
from .interp1_clip import interp1_clip

def do_srgb_gam(inp):
    srgb_lut = get_srgb_lut()
    x = np.linspace(0, 1, 256)
    return interp1_clip(x, srgb_lut, inp)
